frontend/src/pages/ApplicationsPage.tsx

```diff
@@ -45,19 +48,26 @@ const ApplicationsPage: React.FC<ApplicationsPageProps> = ({
   emptyMessage,
   emptyStatePage,
   headerAction,
+  headerContent,
   provideChildrenPadding,
 }) => {
   const renderHeader = () => (
     <PageSection variant={PageSectionVariants.light}>
-      <Split>
-        <SplitItem isFilled>
-          <TextContent>
-            <Text component="h1">{title}</Text>
-            <Text component="p">{description}</Text>
-          </TextContent>
-        </SplitItem>
-        {headerAction && <SplitItem>{headerAction}</SplitItem>}
-      </Split>
+      <Stack hasGutter>
+        <StackItem>
+          <Split>
+            <SplitItem isFilled>
+              <TextContent>
+                <Text component="h1">{title}</Text>
+                <Text component="p">{description}</Text>
+              </TextContent>
+            </SplitItem>
+            {headerAction && <SplitItem>{headerAction}</SplitItem>}
+          </Split>
+        </StackItem>
+        {headerContent && <StackItem>{headerContent}</StackItem>}
+      </Stack>
+      {/* Deprecated */}
       {toolbar}
```

Comment:
This should be removed in the work Alex is doing -- I left it here so we can clean it up later. Used by Metrics (disabled in the pipelines branch).

`headerContent` is the name I chose for the replacement prop -- it covers the same use-case, but does not assume structure and uses the stack component to space everything.